File: Week2/김다빈/BOJ_1969_4/1969.py
n, m = map(int, input().split())
dnaArr = [input() for _ in range(n)]
hammingDistance = 0
dnaName = ""
dnaInfo = ["A", "C", "G", "T"]

for i in range(m):
    curDna = ""
    num = n + 1

    for j in range(4):
        curNum = 0
        for k in range(n):
            if dnaInfo[j] != dnaArr[k][i]:
                curNum += 1

        if curNum < num:
            num = curNum
            curDna = dnaInfo[j]

    dnaName += curDna
    hammingDistance += num


print(dnaName)
print(hammingDistance)

File: Week2/김다빈/BOJ_1969_4/1969.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Scanner;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());

        String[] dnaArr = new String[n];
        for (int i = 0; i < n; i++) {
            dnaArr[i] = br.readLine();
        }

        String[] dnaInfo = {"A", "C", "G", "T"};
        String dnaName = "";
        int hammingDistance = 0;

        for (int i = 0; i < m; i++) {
            String curDna = "";
            int minDiff = n + 1;

            for (String dna : dnaInfo) {
                int diffCount = 0;

                for (int j = 0; j < n; j++) {
                    if (dnaArr[j].charAt(i) != dna.charAt(0)) {
                        diffCount++;
                    }
                }

                if (diffCount < minDiff) {
                    minDiff = diffCount;
                    curDna = dna;
                }
            }

            dnaName += curDna;
            hammingDistance += minDiff;
        }
        System.out.println(dnaName);
        System.out.println(hammingDistance);
    }
}